src/Cpptraj.cpp

```diff
@@ -311,17 +311,19 @@ Cpptraj::Mode Cpptraj::ProcessCmdLineArgs(int argc, char** argv) {
       if (c_err == Command::C_QUIT) return QUIT;
     }
   }
-  // Determine whether to enter interactive mode
-  if (!hasInput || interactive) {
-    // Test if input is really from a console
-    if ( isatty(fileno(stdin)) )
-      return INTERACTIVE;
-    else {
-      // "" means read from STDIN
-      Command::RetType c_err = Command::ProcessInput( State_, "" ); 
-      if (c_err == Command::C_ERR && State_.ExitOnError()) return ERROR;
-      if (c_err == Command::C_QUIT) return QUIT;
-    }
+  // Determine whether to enter interactive mode.
+  if (interactive) {
+    // User explicitly requested ``--interactive``. Do not check isatty.
+    return INTERACTIVE;
+  }
+  if (!hasInput && isatty(fileno(stdin))) {
```

Comment:
I think maybe this needs to be:
```C++
  if (!hasInput) {
    if (isatty(fileno(stdin)))
      return INTERACTIVE;
    else {
      Cmd::RetType c_err = Command::ProcessInput( State_, "" );
      if (c_err == Cmd::ERR && State_.ExitOnError()) return ERROR;
      if (c_err == Cmd::QUIT) return QUIT;
    }
  }
  return BATCH;
```
Otherwise whenever we are not entering batch mode because input is from a terminal, we're automatically trying to get input from a STDIN redirect, and therefore never getting down to BATCH right?